src/core/ext/transport/chttp2/server/chttp2_server.cc

```diff
@@ -1076,12 +1080,25 @@ void grpc_server_add_channel_from_fd(grpc_server* server, int fd,
 
   grpc_core::ChannelArgs server_args = core_server->channel_args();
   std::string name = absl::StrCat("fd:", fd);
+  // The fd is going to be registered with the gRPC poller to be asynchronously
+  // notified for read/write/error events. It must be marked as NonBlocking
+  // to ensure that the poller/gRPC executor threads do not get blocked on
+  // the socket when attempting to read/write from it.
+  grpc_core::PosixSocketWrapper sock(fd);
+  (void)sock.SetSocketNonBlocking(true);
+  (void)sock.SetSocketCloexec(true);
+  (void)sock.SetSocketNoSigpipeIfPossible();
+  auto peer_address = sock.PeerAddress();
+  std::string addr_str = "";
+  if (peer_address.ok()) {
+    addr_str = *grpc_core::ResolvedAddressToURI(peer_address.value());
+  }
   auto memory_quota =
       server_args.GetObject<grpc_core::ResourceQuota>()->memory_quota();
   grpc_endpoint* server_endpoint = grpc_tcp_create_from_fd(
       grpc_fd_create(fd, name.c_str(), true),
       grpc_event_engine::experimental::ChannelArgsEndpointConfig(server_args),
-      name);
+      addr_str);
```

Comment:
If there's an issue getting the peer address, this will be "" instead of "fd:N". Should we fall back to `name` in that case?

Reply:
Fixed. it now.